src/Linux/Avalonia.LinuxFramebuffer/LinuxFramebufferPlatform.cs

```diff
@@ -152,5 +157,79 @@ public static int StartLinuxDirect<T>(this T builder, string[] args, IOutputBack
         builder.Instance.Run(lifetime.Token);
         return lifetime.ExitCode;
     }
+
+    public static DrmOutput CreateDrmOutput<T>(this T builder, string path = null, bool connectorsForceProbe = false,
+        [CanBeNull] DrmOutputOptions options = null)
+        where T : AppBuilderBase<T>, new() => CreateDrmOutput(path, connectorsForceProbe, options);
+    
+    private static DrmOutput CreateDrmOutput(string path = null, bool connectorsForceProbe = false,
+        [CanBeNull] DrmOutputOptions options = null)
+    {
+        DrmCard card = null;
+        DrmResources resources = null;
+        
+        if (!string.IsNullOrEmpty(path))
+        {
+            if (TryCreateDrmOutputForCard(path, out var drmCard, out var drmResources, connectorsForceProbe))
+            {
+                card = drmCard;
+                resources = drmResources;
+            }
+        }
+        else
+        {
+            var files = Directory.GetFiles("/dev/dri/")
+                .Where(w => Regex.Match(w, "card[0-9]+").Success)
+                .OrderBy(o => o);
+            foreach(var file in files) 
+            {
+                if (TryCreateDrmOutputForCard(path, out var drmCard, out var drmResources, connectorsForceProbe))
+                {
+                    card = drmCard;
+                    resources = drmResources;
+                    break;
+                } 
+            }
+        }
+
+        if (card == null || resources == null)
+            throw new InvalidOperationException(string.IsNullOrEmpty(path) ?
+                "Unable to auto-detect an DrmCard" : $"Unable to open DrmCard or create DrmResources for {path}");
+        
+        var connector =
```

Comment:
Since we are auto-detecting the first available output, connector's check should happen in `TryCreateDrmOutputForCard`

Reply:
I agree